recorder.go

```diff
@@ -371,9 +371,18 @@ func convertToKeyValue(k string, value interface{}) *cpb.KeyValue {
 	case bool:
 		kv.Value = &cpb.KeyValue_BoolValue{v}
 	default:
-		glog.Infof("value: %v, %T, is an unsupported type, and has been converted to string", v, v)
-		// TODO: use reflection so that not all custom types have to be converted to string
-		kv.Value = &cpb.KeyValue_StringValue{fmt.Sprint(v)}
+		r := reflect.ValueOf(v)
+		r = reflect.Indirect(r)
```

Comment:
I think this will be fine, but can you appease my neuroses by adding tests for 
```go
convertToKeyValue("k", nil)
```
and
```go
var p *int
convertToKeyValue("k", p)
```